### PR TITLE
Provide an efficient way to lookup wallet by ID #394

### PR DESCRIPTION
**Pull request by adecaro**
Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: integration/token/fungible/topology.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func Topology(backend string, tokenSDKDriver string, auditorAsIssuer bool) []api
 
 	// FSC
 	fscTopology := fsc.NewTopology()
-	//fscTopology.SetLogging("debug", "")
+	fscTopology.SetLogging("debug", "")
```
Comment (HagarMeir) [inline integration/token/fungible/topology.go:41]: comment out please

File: token/core/fabtoken/service.go [hunk 2]
```diff
@@ -8,9 +8,11 @@ package fabtoken
 
 import (
 	"fmt"
-	"sync"
+
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
```
Comment (HagarMeir) [inline token/core/fabtoken/service.go:13]: remove extra line

File: token/core/fabtoken/wallet.go [hunk 3]
```diff
@@ -165,38 +155,31 @@ func (s *Service) AuditorWalletByIdentity(id view.Identity) driver.AuditorWallet
...
-	idInfo, err := s.IP.GetIdentityInfo(driver.AuditorRole, walletID)
-	if err != nil {
-		logger.Errorf("failed to get auditor wallet info for [%s:%s]: %s", walletID, identity.String(), err)
-		return nil
-	}
-
 	idInfoIdentity, _, err := idInfo.Get()
 	if err != nil {
-		logger.Errorf("failed to get auditor wallet identity for [%s:%s]", walletID, identity.String())
-		return nil
```
Comment (HagarMeir) [inline token/core/fabtoken/wallet.go:194]: Missing `return nil`

File: token/core/identity/registry_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package identity
+
+import (
+	"testing"
+
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+
```
Comment (HagarMeir) [inline token/core/identity/registry_test.go:13]: extra line

File: token/core/identity/registry.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+		}
+		if err == nil && len(identityWID) != 0 {
+			w, ok := r.Wallets[identityWID]
+			if ok {
+				if logger.IsEnabledFor(zapcore.DebugLevel) {
+					logger.Debugf("found wallet [%s:%s:%s:%s]", identity, walletID, w.Wallet.ID(), identityWID)
+				}
+				return w.Wallet, nil, identityWID, nil
+			}
+		} /*else {
```
Comment (HagarMeir) [inline token/core/identity/registry.go:82]: please remove if not needed

File: token/core/identity/registry.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+	idInfo, err := r.IdentityProvider.GetIdentityInfo(r.IdentityRole, walletID)
+	if err != nil {
+		return nil, nil, wID, errors.WithMessagef(err, "failed to get wwallet info for [%s]", walletID)
+	}
+
+	return nil, idInfo, wID, nil
+}
+
+// RegisterWallet binds the passed wallet to the passed it
```
Comment (HagarMeir) [inline token/core/identity/registry.go:104]: passed **id**

File: token/core/identity/registry.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+func (r *WalletsRegistry) RegisterWallet(id string, w driver.Wallet) {
+	r.Wallets[id] = &WalletEntry{
+		Prefix: fmt.Sprintf("%s-%s-%s-%s", r.ID.Network, r.ID.Channel, r.ID.Namespace, id),
+		Wallet: w,
+	}
+}
+
+// RegisterIdentity binds the passed identity to the passed wallet identifier
+func (r *WalletsRegistry) RegisterIdentity(identity view.Identity, wID string) error {
+	if logger.IsEnabledFor(zapcore.DebugLevel) {
```
Comment (HagarMeir) [inline token/core/identity/registry.go:114]: Why do you need this here?

File: token/core/identity/registry.go [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+	return nil
+}
+
+// GetWallet returns the wallet identifier bound to the passed identity
+func (r *WalletsRegistry) GetWallet(identity view.Identity) (string, error) {
+	var wID string
+	if err := r.KVS.Get(identity.Hash(), &wID); err != nil {
+		return "", err
+	}
+	if logger.IsEnabledFor(zapcore.DebugLevel) {
```
Comment (HagarMeir) [inline token/core/identity/registry.go:133]: Why is this needed here?

File: token/core/fabtoken/wallet.go [hunk 9]
```diff
@@ -165,38 +155,31 @@ func (s *Service) AuditorWalletByIdentity(id view.Identity) driver.AuditorWallet
...
+		logger.Errorf("failed to get auditor wallet identity for [%s:%s:%s]: %s", wID, id, err)
```
Comment (KElkhiyaoui) [inline token/core/fabtoken/wallet.go:174]: auditor wallet identity for [%s, %s]:%s

File: token/core/zkatdlog/nogh/wallet.go [hunk 10]
```diff
@@ -66,135 +65,112 @@ func (s *Service) Wallet(identity view.Identity) api2.Wallet {
...
-	idInfo, err := s.identityProvider.GetIdentityInfo(api2.AuditorRole, walletID)
-	if err != nil {
-		logger.Errorf("failed to get auditor wallet identity info for [%s:%s:%s]: %s", walletID, identity.String(), id, err)
-		return nil
-	}
-
 	idInfoIdentity, _, err := idInfo.Get()
 	if err != nil {
-		logger.Errorf("failed to get auditor wallet identity for [%s:%s:%s]: %s", walletID, identity.String(), id, err)
+		logger.Errorf("failed to get auditor wallet identity for [%s:%s:%s]: %s", wID, id, err)
```
Comment (KElkhiyaoui) [inline token/core/zkatdlog/nogh/wallet.go:158]: an extra %s
